peep.c

```diff
@@ -3379,8 +3376,7 @@ Perl_rpeep(pTHX_ OP *o)
                             && (p->op_private & OPpLVAL_INTRO) == intro
                             && !(p->op_private & ~OPpLVAL_INTRO)
                             && p->op_next
-                            && (   p->op_next->op_type == OP_NEXTSTATE
-                                || p->op_next->op_type == OP_DBSTATE)
```

Comment:
Changed from checking `p->op_next` to `p`.

Reply:
Oops, so it is, good catch. Will force-push

Reply:
Failing tests caught it before I noticed it :)